### PR TITLE
chore(admission): improve webhook documentation

### PR DESCRIPTION
**Pull request by wdhif**
### What does this PR do?

Improves the `admission` package `doc.go` files.

### Motivation

Our `doc.go` files should be as clear and concise as possible.

### CODE REVIEW

File: pkg/clusteragent/admission/doc.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+// Here's a brief description of each function and what they are used for:
+// - Name: it's the name of the webhook. It's used to identify it. The name
+// appears in some telemetry tags.
+// - WebhookType: it's the type of the webhook. It can be either "mutating" or
+// "validating".
+// - IsEnabled: returns whether the webhook is enabled or not. In general, the
+// recommendation is to disable the webhook by default unless it's needed by a
+// core feature that should be enabled for everyone that deploys the Datadog
+// Agent on Kubernetes.
+// - Endpoint: the endpoint where the webhook is registered.
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:36]:
```suggestion
// - Endpoint: The endpoint where the webhook is registered.
```

File: pkg/clusteragent/admission/doc.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+// - Resources: the Kubernetes resources that the webhook is interested in. For
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:37]:
```suggestion
// - Resources: The Kubernetes resources that the webhook is interested in. For
```

File: pkg/clusteragent/admission/doc.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+// example, pods, deployments, etc.
+// - Operations: the operations applied to the resources that the webhook is
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:39]:
```suggestion
// - Operations: The operations applied to the resources that the webhook is
```

File: pkg/clusteragent/admission/doc.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+// - WebhookFunc: the function that runs the logic of the webhook and returns the admission response.
+//
+// As any other feature, webhooks can be configured using the Datadog
+// configuration. When adding new configuration parameters, please try to follow
+// the convention of the other webhooks. The configuration parameters
+// for a webhook should be under the "admission_controller.name_of_the_webhook"
+// key.
+//
+// Webhooks emit telemetry metrics. Each webhook can define its own
+// metrics as needed but some metrics like "webhooks_received" are common
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:58]:
```suggestion
// metrics as needed, but some metrics like "webhooks_received" are common
```

File: pkg/clusteragent/admission/doc.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+// to all webhooks and defined in common code, so new webhooks can use
+// them without having to define them again.
+//
+// When implementing a new webhook keep performance in mind. For instance, if
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:62]:
```suggestion
// When implementing a new webhook, keep performance in mind. For example, if
```

File: pkg/clusteragent/admission/doc.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+// For example, we can configure the webhook to only receive requests about pods
+// that have a specific label. For performance reasons, we should try to
+// minimize the number of requests that the webhook receives. The label
+// selectors help us with that. There are some default label selectors defined
+// in the "common" package.
+// - MatchConditions: allow us to filter the requests that the webhook receives using
+// a more fine-grained approach than label selectors. This uses the CEL language.
+// - WebhookFunc: the function that runs the logic of the webhook and returns the admission response.
+//
+// As any other feature, webhooks can be configured using the Datadog
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:51]:
```suggestion
// As with any other feature, webhooks can be configured using the Datadog
```

File: pkg/clusteragent/admission/doc.go [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+// configuration. When adding new configuration parameters, please try to follow
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:52]:
```suggestion
// configuration. When adding new configuration parameters, try to follow
```

File: pkg/clusteragent/admission/doc.go [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+// interested in. For example: create, update, delete, etc.
+// - LabelSelectors: allow us to filter the requests that the webhook receives.
+// For example, we can configure the webhook to only receive requests about pods
+// that have a specific label. For performance reasons, we should try to
+// minimize the number of requests that the webhook receives. The label
+// selectors help us with that. There are some default label selectors defined
+// in the "common" package.
+// - MatchConditions: allow us to filter the requests that the webhook receives using
+// a more fine-grained approach than label selectors. This uses the CEL language.
+// - WebhookFunc: the function that runs the logic of the webhook and returns the admission response.
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:49]:
```suggestion
// - WebhookFunc: The function that runs the logic of the webhook and returns the admission response.
```

File: pkg/clusteragent/admission/doc.go [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+// - Operations: the operations applied to the resources that the webhook is
+// interested in. For example: create, update, delete, etc.
+// - LabelSelectors: allow us to filter the requests that the webhook receives.
+// For example, we can configure the webhook to only receive requests about pods
+// that have a specific label. For performance reasons, we should try to
+// minimize the number of requests that the webhook receives. The label
+// selectors help us with that. There are some default label selectors defined
+// in the "common" package.
+// - MatchConditions: allow us to filter the requests that the webhook receives using
+// a more fine-grained approach than label selectors. This uses the CEL language.
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:48]:
```suggestion
// - MatchConditions:  Enables you to filter the requests received by the webhook with a more fine-grained approach than label selectors, using the CEL language.
```

File: pkg/clusteragent/admission/doc.go [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+// - IsEnabled: returns whether the webhook is enabled or not. In general, the
+// recommendation is to disable the webhook by default unless it's needed by a
+// core feature that should be enabled for everyone that deploys the Datadog
+// Agent on Kubernetes.
+// - Endpoint: the endpoint where the webhook is registered.
+// - Resources: the Kubernetes resources that the webhook is interested in. For
+// example, pods, deployments, etc.
+// - Operations: the operations applied to the resources that the webhook is
+// interested in. For example: create, update, delete, etc.
+// - LabelSelectors: allow us to filter the requests that the webhook receives.
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:41]:
```suggestion
// - LabelSelectors: Enables you to filter the requests that the webhook receives.
```

Comment (drichards-87) [reply pkg/clusteragent/admission/doc.go:41]: I'm not sure if "us" in this context refers to Datadog or to the customer. If it refers to the customer, then we should change "us" to "you".

File: pkg/clusteragent/admission/doc.go [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+// Each webhook needs to implement the "Webhook" interface.
+// Here's a brief description of each function and what they are used for:
+// - Name: it's the name of the webhook. It's used to identify it. The name
+// appears in some telemetry tags.
+// - WebhookType: it's the type of the webhook. It can be either "mutating" or
+// "validating".
+// - IsEnabled: returns whether the webhook is enabled or not. In general, the
+// recommendation is to disable the webhook by default unless it's needed by a
+// core feature that should be enabled for everyone that deploys the Datadog
+// Agent on Kubernetes.
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:35]:
```suggestion
// - IsEnabled: Indicates whether the webhook is enabled. In general, the
// recommendation is to disable the webhook by default, unless it's needed by a
// core feature that should be enabled for everyone that deploys the Datadog
// Agent on Kubernetes.
```

File: pkg/clusteragent/admission/doc.go [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+// - Resources: the Kubernetes resources that the webhook is interested in. For
+// example, pods, deployments, etc.
+// - Operations: the operations applied to the resources that the webhook is
+// interested in. For example: create, update, delete, etc.
+// - LabelSelectors: allow us to filter the requests that the webhook receives.
+// For example, we can configure the webhook to only receive requests about pods
+// that have a specific label. For performance reasons, we should try to
+// minimize the number of requests that the webhook receives. The label
+// selectors help us with that. There are some default label selectors defined
+// in the "common" package.
```
Comment (drichards-87) [inline pkg/clusteragent/admission/doc.go:46]:
```suggestion
// For example, you can configure the webhook to only receive requests about pods
// that have a specific label. For performance reasons, try to
// minimize the number of requests that the webhook receives. The label
// selectors help with that. There are some default label selectors defined
// in the "common" package.
```

**Review by drichards-87 — APPROVED** [review]
Left some suggestions from Docs and approved the PR.

File: pkg/clusteragent/admission/doc.go (hunk 10 above)
Comment (wdhif) [reply pkg/clusteragent/admission/doc.go:41]: It's for Datadog, but I'm thinking that "you" is also fine in this context as it's a tool for other teams to use.

File: pkg/clusteragent/admission/doc.go (hunk 9 above)
Comment (wdhif) [reply pkg/clusteragent/admission/doc.go:48]: There was a double-space in the suggestion so I copied it 👍

File: pkg/clusteragent/admission/doc.go [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+// example, pods, deployments, etc.
+// - Operations: The operations applied to the resources that the webhook is
+// interested in. For example: create, update, delete, etc.
+// - LabelSelectors: Enables you to filter the requests that the webhook receives.
+// For example, you can configure the webhook to only receive requests about pods
+// that have a specific label. For performance reasons, try to
+// minimize the number of requests that the webhook receives. The label
+// selectors help with that. There are some default label selectors defined
+// in the "common" package.
+// - MatchConditions: Enables you to filter the requests received by the webhook with a more fine-grained approach than label selectors, using the CEL language.
```
Comment (wdhif) [inline pkg/clusteragent/admission/doc.go:47]:
```suggestion
// - MatchConditions: Enables you to filter the requests received by the webhook
// with a more fine-grained approach than label selectors, using the CEL language.
```

**Review by zhuminyi — APPROVED** [review]

**Pull request merged**